cli/cmd/packageInfo.go

```diff
@@ -0,0 +1,248 @@
+				color.Red("Please either specify the .sig file (Graphene) or SGX enclave binary (Occlum / SGX SDK) directly, or the root directory of an Graphene or Occlum instance.")
+			}
+
+			fmt.Printf("\n")
+			// Output exact errors for each detection method to user
+			if errOpenEnclave != nil {
+				color.Red("Open Enclave detection error: %v\n", errOpenEnclave)
+			}
+			fmt.Printf("Error - Graphene: %v\n", errGraphene)
+			fmt.Printf("Error - Occlum / SGX SDK: %v\n", errOcclum)
+			return errors.New("unable to determine enclave properties")
+		},
+		SilenceUsage: true,
+	}
+
+	return cmd
+}
+
+func decodeSGXSDKSigStruct(path string, isDirectory bool) error {
+	// If the path is a directory, we try to find out if it's an Occlum image directory
+	var elfFile *elf.File
+	var isOcclumInstance bool
+	var err error
+	if isDirectory {
+		if elfFile, err = elf.Open(filepath.Join(path, "build/lib/libocclum-libos.signed.so")); err == nil {
+			color.Green("Detected Occlum image.")
+			isOcclumInstance = true
+		} else if os.IsNotExist(err) {
+			return err
+		}
+	} else {
+		elfFile, err = elf.Open(path)
+	}
+	if err != nil {
+		return err
+	}
+	defer elfFile.Close()
+
+	// SIGSTRUCT is stored in ELF section '.note.sgxmeta'
+	sgxMetaSection := elfFile.Section(".note.sgxmeta")
+	if sgxMetaSection == nil {
+		return errors.New("could not find SGX metadata section (.note.sgxmeta) in given file")
+	}
+
+	sgxMetaData, err := sgxMetaSection.Data()
+	if err != nil {
+		return err
+	}
+
+	// Parse and retrieve values from SIGSTRUCT
+	mrenclave, mrsigner, isvprodid, isvsvn, err := parseSigStruct(sgxMetaData)
+	if err != nil {
+		return nil
+	}
+
+	// Display the determined properties
+	if isOcclumInstance {
+		color.Cyan("PackageProperties for Occlum image at '%s':\n", path)
+	} else {
+		color.Cyan("PackageProperties for '%s':\n", path)
+	}
+
+	printPackageProperties(mrenclave, mrsigner, isvprodid, isvsvn)
+
+	os.Exit(0)
+	return nil
+}
+
+func parseSigStruct(sgxMetaData []byte) ([]byte, []byte, []byte, []byte, error) {
+	/*
+	 * From the "Intel(r) 64 and IA-32 Architectures Software Developer
+	 * Manual, Volume 3: System Programming Guide", Chapter 38, Section 13,
+	 * Table 38-19 "Layout of Enclave Signature Structure (SIGSTRUCT)"
+	 */
+	sigStructHeader := []byte{0x06, 0x00, 0x00, 0x00, 0xe1, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00}
+
+	sigStructIndex := bytes.Index(sgxMetaData, sigStructHeader)
+	if sigStructIndex == -1 {
+		return nil, nil, nil, nil, errors.New("could not find SIGSTRUCT header in given file")
+	}
+
+	// The Intel Software Developer Manual specifies SIGSTRUCT entries up to 1808 bytes.
+	// We use this as a cutoff for our sigStruct slice.
+	if len(sgxMetaData) <= sigStructIndex+1807 {
+		return nil, nil, nil, nil, errors.New("SGX metadata/SIGSTRUCT appears to be too small")
+	}
+
+	sigStruct := sgxMetaData[sigStructIndex : sigStructIndex+1807]
```

Comment:
```suggestion
	const sigStructLen = 1808
	if len(sgxMetaData) < sigStructIndex+sigStructLen {
		return nil, nil, nil, nil, errors.New("SGX metadata/SIGSTRUCT appears to be too small")
	}

	sigStruct := sgxMetaData[sigStructIndex : sigStructIndex+sigStructLen]
```
I think the error was only in the comparison because the slicing excludes the last element.